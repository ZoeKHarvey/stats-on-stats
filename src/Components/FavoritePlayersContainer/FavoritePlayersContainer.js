import React from 'react';
import FavoriteCard from '../FavoriteCard/FavoriteCard';
import { connect } from 'react-redux';
import './FavoritePlayersContainer.scss';
import PropTypes from 'prop-types'

export const FavoritePlayersContainer = ({teamSchedule, getSinglePlayer, favoritePlayers}) => {
  const player = favoritePlayers.map((player) => {
    return <FavoriteCard key={Date.now()}
                        {...player}
                        getSinglePlayer = { getSinglePlayer } 
                        teamSchedule = { teamSchedule }
                        />
});

return(
  <section className="favorite__player--container">
  <h1>Favorite Players</h1>
  {player}
  </section>
)
};

export const mapStateToProps = (state) => ({
  roster: state.roster,
  teamSchedule: state.teamSchedule,
  favoritePlayers: state.favoritePlayers
});

export default connect(mapStateToProps)(FavoritePlayersContainer)

FavoriteCard.propTypes = {
  roster: PropTypes.object,
  teamSchedule: PropTypes.object,
  favoritePlayers: PropTypes.array
};