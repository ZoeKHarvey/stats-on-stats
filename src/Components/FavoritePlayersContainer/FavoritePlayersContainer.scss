.favorite__player--container{
  background-image: url(../../images/ice-texture.jpeg);
  height: 85vh;
  text-align:center;
}