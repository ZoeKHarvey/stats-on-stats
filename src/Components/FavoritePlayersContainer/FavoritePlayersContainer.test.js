import React from 'react';
import { FavoritePlayersContainer, mapStateToProps } from './FavoritePlayersContainer';
import { shallow } from 'enzyme';

describe('FavoitePlayersContainer', () => {
  let wrapper;
  let favoritePlayers = [{name: 'name'}]

  beforeEach(() => {
    wrapper = shallow(
      <FavoritePlayersContainer 
         favoritePlayers={favoritePlayers}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mockStateToProps', () => {
  it('should return an object with the roster data array', () => {
    const mockState = {
      roster: [{jerseyNumber: 1}, {jerseyNumber: 2}],
      teamSchedule: {home: 'here', away: 'there'},
      favoritePlayers: {name: 'pants'}
    };
    const expected = {
      roster: [{jerseyNumber: 1}, {jerseyNumber: 2}],
      teamSchedule: {home: 'here', away: 'there'},
      favoritePlayers: {name: 'pants'}
    };
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  })
})