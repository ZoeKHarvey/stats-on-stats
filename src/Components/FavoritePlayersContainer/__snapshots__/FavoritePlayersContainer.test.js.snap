// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoitePlayersContainer should match the snapshot 1`] = `
<section
  className="favorite__player--container"
>
  <h1>
    Favorite Players
  </h1>
  <FavoriteCard
    key="0"
    name="name"
  />
</section>
`;
